test/tools/router_check/coverage.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/api/v2/core/base.pb.h"
+#include "envoy/router/router.h"
+
+#include "common/common/empty_string.h"
+#include "common/common/logger.h"
+#include "common/stats/fake_symbol_table_impl.h"
+
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+class Coverage : Logger::Loggable<Logger::Id::testing> {
+public:
+  Coverage(envoy::api::v2::RouteConfiguration config) : route_config_(config){};
+  void markCovered(const Envoy::Router::RouteEntry* route);
```

Comment:
Pass a const reference as this can't be null

Reply:
I might be missing something but if I use a reference here won't I have comparison issues? I've attempt to implement this and haven't been successful but it's very likely that I'm overlooking something simple.

Reply:
A reference is just a pointer that isn't null. If you want to compare a reference to a pointer all you need to do is take the address of it (`ptr == &ref`).

Reply:
Ah ok. I think I actually found my issue. Simple oversight on my part :)